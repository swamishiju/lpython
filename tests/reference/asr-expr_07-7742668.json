{
    "basename": "asr-expr_07-7742668",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/expr_07.py",
    "infile_hash": "5c3cc1f1662783acd989ceafcb0eb3259e0194dad005c5bd4e502710",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr_07-7742668.stdout",
    "stdout_hash": "0557040393fc03b9df797ecc4f2bb5fba71f49977dbd8dd6d24fe522",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}