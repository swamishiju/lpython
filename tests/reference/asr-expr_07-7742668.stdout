(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [f
                                    bool_to_str]
                                    []
                                    [(SubroutineCall
                                        2 f
                                        ()
                                        []
                                        ()
                                    )
                                    (SubroutineCall
                                        2 bool_to_str
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            bool_to_str:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            var:
                                                (Variable
                                                    5
                                                    var
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Logical 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    bool_to_str
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 5 var)
                                        (LogicalConstant
                                            .true.
                                            (Logical 4)
                                        )
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(Cast
                                                (LogicalConstant
                                                    .true.
                                                    (Logical 4)
                                                )
                                                LogicalToString
                                                (Allocatable
                                                    (String 1 () DeferredLength DescriptorString)
                                                )
                                                (StringConstant
                                                    "True"
                                                    (Allocatable
                                                        (String 1 () DeferredLength DescriptorString)
                                                    )
                                                )
                                            )]
                                            FormatPythonFormat
                                            (Allocatable
                                                (String 1 () DeferredLength DescriptorString)
                                            )
                                            ()
                                        )
                                    )
                                    (Assert
                                        (StringCompare
                                            (Cast
                                                (Var 5 var)
                                                LogicalToString
                                                (Allocatable
                                                    (String 1 () DeferredLength DescriptorString)
                                                )
                                                ()
                                            )
                                            Eq
                                            (StringConstant
                                                "True"
                                                (String 1 (IntegerConstant 4 (Integer 4) Decimal) ExpressionLength DescriptorString)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 5 var)
                                        (LogicalConstant
                                            .false.
                                            (Logical 4)
                                        )
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Assert
                                        (StringCompare
                                            (Cast
                                                (Var 5 var)
                                                LogicalToString
                                                (Allocatable
                                                    (String 1 () DeferredLength DescriptorString)
                                                )
                                                ()
                                            )
                                            Eq
                                            (StringConstant
                                                "False"
                                                (String 1 (IntegerConstant 5 (Integer 4) Decimal) ExpressionLength DescriptorString)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (Cast
                                                (LogicalConstant
                                                    .true.
                                                    (Logical 4)
                                                )
                                                LogicalToString
                                                (Allocatable
                                                    (String 1 () DeferredLength DescriptorString)
                                                )
                                                (StringConstant
                                                    "True"
                                                    (Allocatable
                                                        (String 1 () DeferredLength DescriptorString)
                                                    )
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "True"
                                                (String 1 (IntegerConstant 4 (Integer 4) Decimal) ExpressionLength DescriptorString)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            a:
                                                (Variable
                                                    4
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    4
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    4
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [g]
                                    []
                                    [(Assignment
                                        (Var 4 a)
                                        (IntegerConstant 5 (Integer 4) Decimal)
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Assignment
                                        (Var 4 x)
                                        (IntegerConstant 3 (Integer 4) Decimal)
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Assignment
                                        (Var 4 x)
                                        (IntegerConstant 5 (Integer 4) Decimal)
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Assignment
                                        (Var 4 b)
                                        (IntegerBinOp
                                            (Var 4 x)
                                            Add
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(Var 4 a)
                                            (Var 4 b)]
                                            FormatPythonFormat
                                            (Allocatable
                                                (String 1 () DeferredLength DescriptorString)
                                            )
                                            ()
                                        )
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (Var 4 b)
                                            Eq
                                            (IntegerConstant 6 (Integer 4) Decimal)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (SubroutineCall
                                        2 g
                                        ()
                                        [((IntegerBinOp
                                            (IntegerBinOp
                                                (Var 4 a)
                                                Mul
                                                (Var 4 b)
                                                (Integer 4)
                                                ()
                                            )
                                            Add
                                            (IntegerConstant 3 (Integer 4) Decimal)
                                            (Integer 4)
                                            ()
                                        ))]
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            g:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    g
                                    (FunctionType
                                        [(Integer 4)]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 x)]
                                    [(Print
                                        (StringFormat
                                            ()
                                            [(Var 3 x)]
                                            FormatPythonFormat
                                            (Allocatable
                                                (String 1 () DeferredLength DescriptorString)
                                            )
                                            ()
                                        )
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    (IntegerConstant 7 (Integer 4) Decimal)
                                    (IntegerConstant 7 (Integer 4) Decimal)
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                    ()
                                    .false.
                                    .false.
                                )
                        })
                    __main__
                    ()
                    []
                    .false.
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        7
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    7
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        7 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                )
        })
    []
)
